### PR TITLE
QEMU v8 S-EL2 support with Hafnium

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Depends on https://github.com/OP-TEE/manifest/pull/241

### CODE REVIEW

File: qemu_v8.mk [hunk 1]
```diff
@@ -356,6 +396,21 @@ optee-os: optee-os-common
...
 
+################################################################################
+# Hafnium
+################################################################################
+
+HAFNIUM_EXPORTS = PATH=$(HAFNIUM_PATH)/prebuilts/linux-x64/clang/bin:$(HAFNIUM_PATH)/prebuilts/linux-x64/dtc:$(PATH)
+
+.hafnium_checkout:
+	(cd $(HAFNIUM_PATH) && git submodule init && git submodule update)
+	touch .hafnium_checkout
```
Comment (jforissier) [inline qemu_v8.mk:407]: `touch $@`

File: qemu_v8.mk [hunk 2]
```diff
@@ -356,6 +396,21 @@ optee-os: optee-os-common
...
 optee-os-clean: optee-os-clean-common
 
+################################################################################
+# Hafnium
+################################################################################
+
+HAFNIUM_EXPORTS = PATH=$(HAFNIUM_PATH)/prebuilts/linux-x64/clang/bin:$(HAFNIUM_PATH)/prebuilts/linux-x64/dtc:$(PATH)
+
+.hafnium_checkout:
+	(cd $(HAFNIUM_PATH) && git submodule init && git submodule update)
```
Comment (jforissier) [inline qemu_v8.mk:406]: `git -C $(HAFNIUM_PATH) submodule init --update`

File: qemu_v8.mk [hunk 3]
```diff
@@ -356,6 +396,21 @@ optee-os: optee-os-common
 
 optee-os-clean: optee-os-clean-common
 
+################################################################################
+# Hafnium
+################################################################################
+
+HAFNIUM_EXPORTS = PATH=$(HAFNIUM_PATH)/prebuilts/linux-x64/clang/bin:$(HAFNIUM_PATH)/prebuilts/linux-x64/dtc:$(PATH)
+
+.hafnium_checkout:
```
Comment (jforissier) [inline qemu_v8.mk:405]: `.hafnium_checkout` needs to be added to `.gitignore`; or better create it under `$(OUT_PATH)`?

File: qemu_v8.mk [hunk 4]
```diff
@@ -356,6 +396,21 @@ optee-os: optee-os-common
...
+	(cd $(HAFNIUM_PATH) && git submodule init && git submodule update)
+	touch .hafnium_checkout
+
+hafnium_base: .hafnium_checkout | $(OUT_PATH)
+	$(HAFNIUM_EXPORTS) $(MAKE) -C $(HAFNIUM_PATH) $(HAFNIUM_FLAGS) all
+
+hafnium: hafnium_base
```
Comment (jforissier) [inline qemu_v8.mk:412]: Why is `hafnium_base` needed? Why not depend on the generated binary? I.e.,:
```make
hafnium: $(HAFNIUM_BIN)

$(HAFNIUM_BIN): .hafnium_checkout | $(OUT_PATH)
	$(HAFNIUM_EXPORTS) $(MAKE) -C $(HAFNIUM_PATH) $(HAFNIUM_FLAGS) all
```

**Review by jforissier — COMMENTED** [review]
Comments for " qemu_v8: add support for hafnium as S-EL2"

**Comment by jforissier** [comment]
For "qemu_v8: remove ENABLE_SME_FOR_*=1 from TF_A_FLAGS":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
For "qemu_v8: update manifest for EL3 SPMC":
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: qemu_v8.mk (hunk 3 above)
Comment (jenswi-linaro) [reply qemu_v8.mk:405]: `$(OUT_PATH)` doesn't seem right, I'd rather keep it in the build or `$(ROOT)` directory. I'll add it to `.gitignore`.

File: qemu_v8.mk (hunk 4 above)
Comment (jenswi-linaro) [reply qemu_v8.mk:412]: Good idea, I'll update.

**Comment by jenswi-linaro** [comment]
"qemu_v8: add support for Hafnium as S-EL2" and tags added to the other two commits.

**Comment by jforissier** [comment]
For "qemu_v8: add support for Hafnium as S-EL2":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Pull request merged**